### PR TITLE
Incorrect location of 'async' for protocol in actor's fix suggestion

### LINKED ISSUE #69244: Incorrect location of 'async' for protocol in actor's fix suggestion
The following code provides a wrong `fix` suggestion:
```swift
protocol P {
    func foo() -> Void
}
actor A: P {
    func foo() { }
}
```
The compiler suggests as fix: `Mark the protocol requirement 'foo()' 'async' to allow actor-isolated conformances` but it adds the `async` to the wrong location:
```swift
func foo() -> async Void // new-error: 'async' may only occur before '->'
```

### PR DESCRIPTION
**Pull request by dancamarotto**
Resolves #69244

### CODE REVIEW

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test

**Review by LucianoPAlmeida — APPROVED** [review]
It looks like this code path was not really covered by tests since nothing broke. So wondering if we could add a few more tests for covering "throws" as well. 
But PR looks good, someone from Apple team have the final say before merging :)

**Comment by LucianoPAlmeida** [comment]
What I meant by throws tests were something like this 

```swift
import Distributed

protocol P {
    func foo() -> Void // here fix-it is also wrong result in func foo() -> async throws Void
}

distributed actor A: P {
    func foo() { }
}
```

In this example the fix-it covers `throws` modifier as well which at the moment is also wrong `func foo() -> async throws Void`. Note that this test since it requires distributed it has to be on a file that has `// REQUIRES: distributed` you can search across for examples.

**Comment by dancamarotto** [comment]
@LucianoPAlmeida ah ok, thanks, I hadn't think of distributed actors - done. please let me know if it does not look correct.

I tried to add the test so that the only error would be what I wanted to test, hence the `typealias ActorSystem` and the `distributed func`.

**Comment by LucianoPAlmeida** [comment]
> @LucianoPAlmeida ah ok, thanks, I hadn't thought of distributed actors - done. please let me know if it does not look correct.

Distributed is the only case where we exercise this `async throws` fix-it, and test does look right. But I don't recommend adding distributed test to this file. This test should probably be in a file that has `// REQUIRES: distributed` otherwise I think it would fail on CI. 

Maybe add that test case to a file under `test/Distributed`? The `test/Distributed/actor_protocols.swift` file seems like a good candidate, but I am no expert in this so my suggestions should be taken with a grain of salt. But let's add it and hear from the final reviewers :)

File: test/decl/class/actor/conformance.swift [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 // REQUIRES: concurrency
 
+import Distributed
```
Comment (LucianoPAlmeida) [inline test/decl/class/actor/conformance.swift:5]: See the comment about adding distributed to this file. For organization purposes, I think there is a better place.

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
This looks good AFAICS, thank you

File: test/decl/class/actor/conformance.swift [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 // REQUIRES: concurrency
```
Comment (ktoso) [inline test/decl/class/actor/conformance.swift:4]:
```suggestion
// REQUIRES: distributed
```

Comment (ktoso) [reply test/decl/class/actor/conformance.swift:4]: I think you need to add that

File: test/decl/class/actor/conformance.swift (hunk 1 above)
Comment (ktoso) [reply test/decl/class/actor/conformance.swift:5]: Yeah this is ok I think though you need to add the REQUIRES: distributed now; i commented above

File: test/decl/class/actor/conformance.swift (hunk 2 above)
Comment (dancamarotto) [reply test/decl/class/actor/conformance.swift:4]: @ktoso @LucianoPAlmeida I've moved the test to `decl/protocol/special/DistributedActor.swift` where it already had a test with distributed actor implementing a protocol and all the required imports.

Comment (ktoso) [reply test/decl/class/actor/conformance.swift:4]: Lookin good, thank you

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: test/decl/class/actor/conformance.swift (hunk 1 above)
Comment (LucianoPAlmeida) [inline test/decl/class/actor/conformance.swift:5]: Now that we moved distributed test to another file, we can revert this import

Comment (dancamarotto) [reply test/decl/class/actor/conformance.swift]: good catch - done.

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please smoke test

**Review by ktoso — APPROVED** [review]

**Pull request merged**